geode-core/src/distributedTest/java/org/apache/geode/internal/cache/RebalanceWhileCreatingRegionDistributedTest.java

```diff
@@ -0,0 +1,298 @@
+    blackboard.initBlackboard();
+
+    // Start Locator
+    MemberVM locator = cluster.startLocatorVM(0);
+
+    // Start servers
+    int locatorPort = locator.getPort();
+    MemberVM server1 = cluster.startServerVM(1, locatorPort);
+    MemberVM server2 = cluster.startServerVM(2, locatorPort);
+    MemberVM accessor = cluster.startServerVM(3, locatorPort);
+
+    // Add DistributionMessageObserver
+    String regionName = testName.getMethodName();
+    Stream.of(server1, server2, accessor)
+        .forEach(server -> server.invoke(() -> addDistributionMessageObserver(regionName)));
+
+    // Create regions in each server
+    InternalDistributedMember source = server1.invoke(() -> {
+      createSingleBucketRegion(regionName, RegionShortcut.PARTITION);
+      Region<Integer, Integer> region =
+          ClusterStartupRule.getCache().getRegion(regionName);
+      region.put(123, 123);
+      PartitionedRegionDataStore partitionedRegionDataStore =
+          ((PartitionedRegion) region).getDataStore();
+      // Make sure server1 has the primary bucket
+      assertThat(partitionedRegionDataStore).isNotNull();
+      assertThat(partitionedRegionDataStore.getNumberOfPrimaryBucketsManaged()).isEqualTo(1);
+      return InternalDistributedSystem.getAnyInstance().getDistributedMember();
+    });
+
+    InternalDistributedMember destination = server2.invoke(() -> {
+      createSingleBucketRegion(regionName, RegionShortcut.PARTITION);
+      Region<Integer, Integer> region =
+          ClusterStartupRule.getCache().getRegion(regionName);
+      PartitionedRegionDataStore partitionedRegionDataStore =
+          ((PartitionedRegion) region).getDataStore();
+      // Make sure server2 does not have primary bucket
+      assertThat(partitionedRegionDataStore).isNotNull();
+      assertThat(partitionedRegionDataStore.getNumberOfPrimaryBucketsManaged()).isEqualTo(0);
+      return InternalDistributedSystem.getAnyInstance().getDistributedMember();
+    });
+
+    // Asynchronously wait to create the proxy region in the accessor
+    AsyncInvocation asyncInvocation = accessor.invokeAsync(() -> {
+      waitToCreateSingleBucketProxyRegion(regionName);
+    });
+
+    // Move the primary bucket from server1 to server2 and close the cache in the end
+    server2.invoke(() -> {
+      PartitionedRegion partitionedRegion =
+          (PartitionedRegion) ClusterStartupRule.getCache().getRegion(regionName);
+      PartitionedRegionDataStore partitionedRegionDataStore = partitionedRegion.getDataStore();
+      partitionedRegionDataStore.moveBucket(0, source, true);
+      ClusterStartupRule.getCache().close();
+    });
+
+    asyncInvocation.get();
+
+    // Make sure the accessor knows that the primary bucket has moved to server2
+    accessor.invoke(() -> {
+      PartitionedRegion pr =
+          (PartitionedRegion) ClusterStartupRule.getCache().getRegion(regionName);
+      assertThat(pr.getRegionAdvisor().getBucket(0).getBucketAdvisor().getProfile(source))
```

Comment:
Adding check to see if the bucket is moved to destination will confirm the bucket has moved successfully

Reply:
Additional check added.